packages/modules/promotion/src/services/promotion-module.ts

```diff
@@ -412,7 +412,15 @@ export default class PromotionModuleService
         }
 
         const attribute = campaignBudget.attribute!
-        const attributeValue = registrationContext[attribute]
+        /**
+         * `promotion_code` is an exception, since it doesn't apply for the entiere context but per promotion
+         * if multiple promotion are applied on a cart, they share computation context
+         */
+        const attributeValue = (
+          attribute === "promotion_code"
+            ? registrationContext[attribute]
+            : promotion.code
+        ) as string | null
```

Comment:
thought: Tbh. I am not a huge fan of this; it feels dirty to have hard-coded exceptions like this in the code. 

I think we should consider adding a limit directly on the promotion as an alternative solution. That limit could be combined with a campaign budget to solve this use case. 

I am curious to hear what you and others think @medusajs/os

Reply:
For context, the use case we are trying to solve here is:
> Some of our users would like to create a single campaign called Customer Service Promos under which their team can create one-time promotions for individual customers. However, the current implementation prevents this because the limit applies at the campaign level rather than per promotion. They are once again left with creating a new campaign for recurring customers (i.e. customers that have already received a code from the campaign in the past).